app/src/main/java/com/duckduckgo/app/sitepermissions/SitePermissionsViewModel.kt

```diff
@@ -0,0 +1,170 @@
+import com.duckduckgo.app.location.data.LocationPermissionsRepository
+import com.duckduckgo.app.settings.db.SettingsDataStore
+import com.duckduckgo.app.sitepermissions.SitePermissionsViewModel.Command.LaunchWebsiteAllowed
+import com.duckduckgo.app.sitepermissions.SitePermissionsViewModel.Command.ShowRemovedAllConfirmationSnackbar
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.site.permissions.impl.SitePermissionsRepository
+import com.duckduckgo.site.permissions.impl.pixels.SitePermissionsPixel.PixelParameter
+import com.duckduckgo.site.permissions.impl.pixels.SitePermissionsPixel.PixelValue
+import com.duckduckgo.site.permissions.impl.pixels.SitePermissionsPixel.SitePermissionsPixelName
+import com.duckduckgo.site.permissions.store.sitepermissions.SitePermissionsEntity
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@ContributesViewModel(ActivityScope::class)
+class SitePermissionsViewModel @Inject constructor(
+    private val sitePermissionsRepository: SitePermissionsRepository,
+    private val locationPermissionsRepository: LocationPermissionsRepository,
+    private val geolocationPermissions: GeoLocationPermissionsManager,
+    private val settingsDataStore: SettingsDataStore,
+    private val dispatcherProvider: DispatcherProvider,
+    private val pixel: Pixel
+) : ViewModel() {
+
+    private val _viewState = MutableStateFlow(ViewState())
+    val viewState: StateFlow<ViewState> = _viewState
+
+    private val _commands = Channel<Command>()
+    val commands: Flow<Command> = _commands.receiveAsFlow()
+
+    data class ViewState(
+        val askLocationEnabled: Boolean = true,
+        val askCameraEnabled: Boolean = false,
+        val askMicEnabled: Boolean = true,
+        val sitesPermissionsAllowed: List<SitePermissionsEntity> = listOf(),
+        val locationPermissionsAllowed: List<LocationPermissionEntity> = listOf()
+    )
+
+    sealed class Command {
+        class ShowRemovedAllConfirmationSnackbar(
+            val removedSitePermissions: List<SitePermissionsEntity>,
+            val removedLocationPermissions: List<LocationPermissionEntity>
+        ) : Command()
+        class LaunchWebsiteAllowed(val domain: String) : Command()
+    }
+
+    init {
+        _viewState.value = ViewState(
+            askLocationEnabled = settingsDataStore.appLocationPermission,
+            askCameraEnabled = sitePermissionsRepository.askCameraEnabled,
+            askMicEnabled = sitePermissionsRepository.askMicEnabled
+        )
+    }
+
+    fun allowedSites() {
+        viewModelScope.launch {
+            locationPermissionsRepository.getLocationPermissionsFlow().collect { locationPermissionsList ->
```

Comment:
Does this work as expected? 😄 I've never tried putting a `collect` inside another `collect`. I'm a bit lost here. Imagine this scenario:
1. we launch the coroutine
2. getLocationPermissionsFlow, we collect first value
3. sitePermissionsWebsitesFlow, we collect first value
4. we emit new values in _viewState.emit(

if `getLocationPermissionsFlow` emits a new value, are we sure we will update the viewstate?

Reply:
You are totally right. There is a [combine](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/combine.html) method which makes much more sense here. Still in a learning curve with coroutines 😅 
I'll change it.